Source/Classes/PINRemoteImageManager.m

```diff
@@ -281,6 +282,16 @@ - (void)setValue:(nullable NSString *)value forHTTPHeaderField:(nullable NSStrin
     });
 }
 
+- (void)setRequestConfiguration:(PINRemoteImageManagerRequestConfigurationHandler)configurationBlock {
```

Comment:
@zachwaugh you're right about the deadlock.

Here's what I would suggest: We need to add a getter which holds the lock when it accesses `requestConfigurationHandler` or else it isn't thread safe to access by clients. However, since we know we have the lock in downloadDataWithURL: we can access the ivar directly instead of using the property.